File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        n = len(prices)
        max_ans = -1
        min_val = float('inf')

        for i in range(n):
            if prices[i] < min_val:
                min_val = prices[i]
            
            if prices[i] - min_val > max_ans:
                max_ans = max(max_ans, prices[i] - min_val)
        return max_ans


        
            
            

File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_ans = INT_MIN;
        int min_val = INT_MAX;

        for (int i = 0; i< prices.size(); i++){
            if (prices[i] < min_val)
                min_val = prices[i];
            
            if ( ( prices[i] - min_val) > max_ans )
                max_ans = max(max_ans, prices[i] - min_val);
        }
        return max_ans;   

    }
};
